.github/workflows/ci-for-default-ns.yaml

```diff
@@ -75,15 +75,18 @@ jobs:
           # for testing start-debug and stop-debug deployment
           # mons
           kubectl rook_ceph debug start rook-ceph-mon-a
-          sleep 5
+          debug_mon=$(kubectl get pods -nrook-ceph | grep rook-ceph-mon-a-debug | awk '{ print $1  }')
+          tests/github-action-helper.sh wait_for_condition "rook-ceph-mon-a-debug deployment pod to start" 10 "kubectl -n rook-ceph get pod $debug_mon | grep 'rook-ceph-mon-a-debug.*Running'"
```

Comment:
This looks good to me calling a simpler one-liner,

But by this, we might be restricting the script implementation to only wait for a pod, which can do a lot of operation waiting in overall, Like waiting for `cephblockpool` etc, I know we don't have many complications in the CI know but in future, it might more help us.....

Just a add-on suggestion otherwise I can go with the above flow:)

Reply:
Currently does the function need to wait for anything except a pod? In this PR that's the only place I see it used. So I'd suggest we optimize for this PR. If a future PR needs to wait for a different resource, a new function could be added to wait for that resource. It will keep the test script a lot cleaner, even if we end up with multiple helper functions for different types of resources.